File: Data Structures & Algorithms/string-encode-and-decode/submission-1.js
class Solution {
    /**
     * @param {string[]} strs
     * @returns {string}
     */
    encode(strs) {
        let encoded = "";
        for(let str of strs)
            encoded += str.length + "#" + str;
        return encoded;
    }

    /**
     * @param {string} str
     * @returns {string[]}
     */
    decode(str) {
        let res = [];
        let i = 0;

        while(i < str.length) {
            // 1. find the next #
            let j = i;
            while(str[j] != "#") j++;

            //the length is between i and j
            let length = parseInt(str.substring(i, j));

            let start = j + 1;
            let end = start + length;
            res.push(str.substring(start, end));

            i = end;
        }

        return res;
    }
}

File: Data Structures & Algorithms/sqrtx/submission-1.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, x
        res = 0

        while l <= r:
            m = l + (r - l) // 2
            square = m * m

            if square > x:
                r = m - 1
            else:
                res = m
                l = m + 1

        return res
